tests/mocha/connection_checker_test.js

```diff
@@ -341,4 +341,117 @@ suite('Connection checker', function() {
+            <block type="text_print" id="T9cdR}mwDn3Xl!*g|mw8" movable="false">
+              <comment pinned="true" h="32" w="88.5333251953125">Immovable</comment>
+            </block>
+          </next>
+        </block>
+        <block type="text_print" id="ZgFlirwB{~}%l/BZ+j?R" x="47" y="-118"/>
+      </xml>`), this.workspace);
+      [this.blockA, this.blockB, this.blockC] = this.workspace.getAllBlocks(true);
+      this.checker = this.workspace.connectionChecker;
+      });
+
+      test('Connect a stack', function() {
+        // block C is not connected to block A; both are movable.
+        chai.assert.isTrue(
+          this.checker.doDragChecks(
+            this.blockC.nextConnection, this.blockA.previousConnection, 9000),
+          'Should connect two compatible stack blocks');
+      });
+
+      test('Do not splice into unmovable stack', function() {
+        // Try to connect blockC above blockB. It shouldn't work because B is not movable
+        // and is already connected to A's nextConnection.
+        chai.assert.isFalse(
+          this.checker.doDragChecks(
+            this.blockC.previousConnection, this.blockA.nextConnection, 9000),
+          'Should not splice in a block above an unmovable block');
+      });
+
+      test('Connect to bottom of unmovable stack', function() {
+        // Try to connect blockC below blockB.
+        // This is allowed even though B is not movable because it is on B's nextConnection.
+        chai.assert.isTrue(
+          this.checker.doDragChecks(
+            this.blockC.previousConnection, this.blockB.nextConnection, 9000),
+          'Should connect below an unmovable stack block');
+      });
+
+      test('Connect to unconnected unmovable block', function() {
+        // Delete blockA.
+        this.blockA.dispose();
+
+        // Try to connect blockC above blockB.
+        // This is allowed because we're not splicing into a stack.
+        chai.assert.isTrue(
+          this.checker.doDragChecks(
+            this.blockC.nextConnection, this.blockB.previousConnection, 9000),
+          'Should connect above an unconnected unmovable block'
+        );
+      });
+    });
+    suite('Rows', function() {
+      setup(function() {
+        this.workspace = Blockly.inject('blocklyDiv');
+        // Load 3 blocks: A and B are connected (input/output); B is unmovable.
+        Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(`<xml xmlns="https://developers.google.com/blockly/xml">
+        <block type="math_single" id="4L?aLhveXpvYaYq.n,!p" x="-87" y="-188">
+          <field name="OP">ROOT</field>
+          <value name="NUM">
+            <block type="math_number" movable="false" id="U)q;b^J7HKl$(5sKtOuT">
+              <field name="NUM">123</field>
+            </block>
+          </value>
+        </block>
+        <block type="math_single" id="u|{o,wAaGRvlFB((U}h=" x="138" y="-188">
+          <field name="OP">ROOT</field>
+        </block>
+      </xml>`), this.workspace);
```

Comment:
Could also use the row test blocks here instead of using the math blocks.

Reply:
Thanks, done!